src/cluster_asm.c

```diff
@@ -272,6 +272,55 @@ size_t asmGetPeakSyncBufferSize(void) {
     return peak;
 }
 
+static inline int asmIsSlotImporting(void) {
+    if (!asmManager || listLength(asmManager->tasks) == 0) return 0;
+
+    /* Only support a single task at a time now, so only check the first task */
+    asmTask *task = listNodeValue(listFirst(asmManager->tasks));
+    /* We only check the destination side, the source side `pauseActions` will
+     * pause the write traffic (including expire/evict). */
+    /* TODO: for failed state, is it safe to expire? cleanup may take long time */
+    if ((task->operation == ASM_IMPORT && task->state != ASM_NONE)) {
+        return 1;
+    }
+
+    return 0;
+}
+
+/* Return true if the given slot supports key expiration.
+ * This is used to determine whether keys in the slot can be
+ * expired or evicted during ASM operations.
+ *
+ * Note: make sure the caller has already checked asmIsSlotImporting() */
+static inline int asmSlotAllowsExpiryOrEvictionCore(int slot) {
+    debugServerAssert(asmIsSlotImporting());
+    asmTask *task = listNodeValue(listFirst(asmManager->tasks));
+    /* TODO: optimize this, use bitmap instead of array */
```

Comment:
maybe we should call these functions like `asmSlotBelongsToThisNode()` which answers why we skip the slot in multiple places? Also, I wonder if we can just check `getNodeBySlot(slot) == getMyClusterNode()` ?

Reply:
make sense, let me think about it

Reply:
I found `clusterNodeCoversSlot` that can be used for this purpose